pkg/console/status/auth_status.go

```diff
@@ -0,0 +1,202 @@
+func (c *AuthStatusHandler) Progressing(reason, message string) *AuthStatusHandler {
+	now := metav1.Now()
+	c.setCondition(ConditionTypeProgressing, metav1.ConditionTrue, reason, message, now)
+	c.setCondition(ConditionTypeDegraded, metav1.ConditionFalse, reason, message, now)
+
+	return c
+}
+
+// Unavailable sets all conditions to False
+func (c *AuthStatusHandler) Unavailable(reason, message string) *AuthStatusHandler {
+	now := metav1.Now()
+	c.setCondition(ConditionTypeAvailable, metav1.ConditionFalse, reason, message, now)
+	c.setCondition(ConditionTypeProgressing, metav1.ConditionFalse, reason, message, now)
+	c.setCondition(ConditionTypeDegraded, metav1.ConditionFalse, reason, message, now)
+
+	return c
+}
+
+// Available sets the Available condition to True, and Progressing and Degraded to False
+func (c *AuthStatusHandler) Available(reason, message string) *AuthStatusHandler {
+	now := metav1.Now()
+	c.setCondition(ConditionTypeAvailable, metav1.ConditionTrue, reason, message, now)
+	c.setCondition(ConditionTypeProgressing, metav1.ConditionFalse, reason, message, now)
+	c.setCondition(ConditionTypeDegraded, metav1.ConditionFalse, reason, message, now)
+
+	return c
+}
+
+func (c *AuthStatusHandler) setCondition(conditionType AuthenticationConditionType, status metav1.ConditionStatus, reason, message string, ts metav1.Time) *AuthStatusHandler {
+	if c.conditionsToApply[conditionType] == nil {
+		c.conditionsToApply[conditionType] = &metav1.Condition{Type: string(conditionType)}
+	}
+
+	c.conditionsToApply[conditionType].Status = status
+	c.conditionsToApply[conditionType].Reason = reason
+	c.conditionsToApply[conditionType].Message = message
+	c.conditionsToApply[conditionType].LastTransitionTime = ts
+
+	return c
+}
+
+func (c *AuthStatusHandler) WithCurrentOIDCClient(currentClientID string) *AuthStatusHandler {
+	c.currentClientID = currentClientID
+	return c
+}
+
+func (c *AuthStatusHandler) Apply(ctx context.Context, authnConfig *configv1.Authentication) error {
+	if authnConfig.Spec.Type != configv1.AuthenticationTypeOIDC {
+		return fmt.Errorf("authentication type not OIDC")
+	}
+
+	applyConfig, err := configv1ac.ExtractAuthenticationStatus(authnConfig, c.fieldManager)
+	if err != nil {
+		return err
+	}
+
+	if applyConfig.Status == nil {
+		applyConfig.Status = &configv1ac.AuthenticationStatusApplyConfiguration{}
+	}
+
+	idx := -1
+	for i, client := range applyConfig.Status.OIDCClients {
+		if *client.ComponentName == c.componentName && *client.ComponentNamespace == c.componentNamespace {
+			idx = i
+			break
+		}
+	}
+
+	// create the OIDCClientStatus for the operator if it doesn't exist
+	if idx == -1 {
+		if applyConfig.Status.OIDCClients == nil {
+			applyConfig.Status.OIDCClients = make([]configv1ac.OIDCClientStatusApplyConfiguration, 0)
+		}
+
+		applyConfig.Status.OIDCClients = append(applyConfig.Status.OIDCClients, configv1ac.OIDCClientStatusApplyConfiguration{
```

Comment:
use the `With*` methods wherever possible

Reply:
aha, TIL :smile: